hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -211,7 +214,24 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
         LOG.debug("Committing new table: {}", fullName);
       }
 
-      tbl.setSd(storageDescriptor(metadata, hiveEngineEnabled)); // set to pickup any schema changes
+      // Pinterest specific: avoid updating the schema for thrift table in the HMS as the schema
+      // stored in HMS tend to have column
+      // type exceed the type string limit set by HMS thus got truncated to become an invalid
+      // schema. When Iceberg tries to update HMS table with
+      // a full size schema, HMS will throw incompatible schema exception. (message:The following
+      // columns have types
+      // incompatible with the existing columns in their respective positions). Since we do not use
+      // HMS as the source of the truth for table schema
+      // (we rely on Iceberg schema stored in Iceberg metadata file which is determined by the
+      // thrift class at the write time), we can safely skip updating schema for thrift table.
+      boolean isThriftBackedTable =
+          !Strings.isNullOrEmpty(metadata.properties().get(TableProperties.THRIFT_TYPE));
+      List<FieldSchema> oldCols = tbl.getSd().getCols();
```

Comment:
I see, the oldCols is the cols published? Probably double check if there will be schema mismatching between the cols and metadata. It most likely should be OK though as we likely just ignore the cols completely in reader side if thrift_type is set.